tests-e2e/cypress/integration/channels/app_bar_spec.js

```diff
@@ -0,0 +1,118 @@
+                    playbookRunName: 'Playbook Run',
+                    ownerUserId: testUser.id,
+                });
+            });
+
+            cy.apiGetConfig(true).then(({config}) => {
+                siteURL = config.SiteURL;
+                appBarFeatureFlagEnabled = config.FeatureFlagAppBarEnabled === 'true';
+            });
+        });
+    });
+
+    beforeEach(() => {
+        // # Size the viewport to show the RHS without covering posts.
+        cy.viewport('macbook-13');
+
+        // # Login as testUser
+        cy.apiLogin(testUser);
+    });
+
+    const getAppBarImageSelector = (siteURL) => `.app-bar .app-bar__icon-inner img[src="${siteURL}/plugins/playbooks/public/app-bar-icon.png"]`;
+
+    describe('App Bar feature flag disabled', () => {
+        it('should not show the Playbook App Bar icon', () => {
+            onlyOn(!appBarFeatureFlagEnabled);
+
+            // # Navigate directly to a non-playbook run channel
+            cy.visit(`/${testTeam.name}/channels/town-square`);
+
+            // * Verify App Bar icon is not showing
+            cy.get('#channel_view').within(() => {
+                cy.get(getAppBarImageSelector(siteURL)).should('not.exist');
+            });
+        });
+    });
+
+    describe('App Bar feature flag enabled', () => {
+        it('should show the Playbook App Bar icon', () => {
+            onlyOn(appBarFeatureFlagEnabled);
+
+            // # Navigate directly to a non-playbook run channel
+            cy.visit(`/${testTeam.name}/channels/town-square`);
+
+            // * Verify App Bar icon is showing
+            cy.get('#channel_view').within(() => {
+                cy.get(getAppBarImageSelector(siteURL)).should('exist');
+            });
+        });
+
+        describe('tooltip text', () => {
+            it('should show "Toggle Playbook List" outside a playbook run channel', () => {
+                onlyOn(appBarFeatureFlagEnabled);
+
+                // # Navigate directly to a non-playbook run channel
+                cy.visit(`/${testTeam.name}/channels/town-square`);
+
+                // # Hover over the channel header icon
+                cy.get(getAppBarImageSelector(siteURL)).trigger('mouseover');
+
+                // * Verify tooltip text
+                cy.get('body > div[role="tooltip"]').contains('Toggle Playbook List');
+            });
+
+            it('should show "Toggle Run Details" inside a playbook run channel', () => {
+                onlyOn(appBarFeatureFlagEnabled);
+
+                // # Navigate directly to a playbook run channel
+                cy.visit(`/${testTeam.name}/channels/playbook-run`);
+
+                // # Hover over the channel header icon
+                cy.get(getAppBarImageSelector(siteURL)).trigger('mouseover');
+                cy.wait(TIMEOUTS.HALF_SEC);
+
+                // * Verify tooltip text
+                cy.get('body > div[role="tooltip"]').contains('Toggle Run Details');
```

Comment:
```suggestion
                cy.findByRole('tooltip', {name: 'Toggle Run Details'}).should('be.visible');
```
I had a bit of flake on this, suggestion seemed to resolve.